packages/pages-components/src/components/schema/yextFields/hours.tsx

```diff
@@ -68,6 +77,111 @@ export const OpeningHoursSchema = (hours?: HoursType) => {
   };
 };
 
+export const OpeningHoursSpecificationSchema = (
+  hours?: HoursType
+): {
+  openingHoursSpecification?: OpeningHoursSpecification[];
+  specialOpeningHoursSpecification?: OpeningHoursSpecification[];
+} => {
+  if (!validateHoursType(hours)) {
+    return {};
+  }
+
+  const mondayHours = getHoursSpecificationByDay(hours.monday, "Monday");
+  const tuesdayHours = getHoursSpecificationByDay(hours.tuesday, "Tuesday");
+  const wednesdayHours = getHoursSpecificationByDay(
+    hours.wednesday,
+    "Wednesday"
+  );
+  const thursdayHours = getHoursSpecificationByDay(hours.thursday, "Thursday");
+  const fridayHours = getHoursSpecificationByDay(hours.friday, "Friday");
+  const saturdayHours = getHoursSpecificationByDay(hours.saturday, "Saturday");
+  const sundayHours = getHoursSpecificationByDay(hours.sunday, "Sunday");
+
+  const openingHoursSpecificationSchema: {
+    openingHoursSpecification?: OpeningHoursSpecification[];
+    specialOpeningHoursSpecification?: OpeningHoursSpecification[];
+  } = {
+    openingHoursSpecification: [
+      ...mondayHours,
+      ...tuesdayHours,
+      ...wednesdayHours,
+      ...thursdayHours,
+      ...fridayHours,
+      ...saturdayHours,
+      ...sundayHours,
+    ],
+  };
+
+  const holidayHoursSpecifications = getHolidayHoursSpecification(
+    hours.holidayHours
+  );
+  if (holidayHoursSpecifications?.length) {
+    openingHoursSpecificationSchema.specialOpeningHoursSpecification =
+      holidayHoursSpecifications;
+  }
+
+  return openingHoursSpecificationSchema;
+};
+
+const getHoursSpecificationByDay = (
+  hours: DayType | undefined,
+  day?: string
```

Comment:
Probably want a type so an arbitrary string can't be used. Then again it's internal so not a big deal, so up to you.